sklearn/linear_model/coordinate_descent.py

```diff
@@ -472,6 +475,10 @@ def enet_path(X, y, l1_ratio=0.5, eps=1e-3, n_alphas=100, alphas=None,
                           ' You might want' +
                           ' to increase the number of iterations',
                           ConvergenceWarning)
+            if X.dtype == np.float32:
```

Comment:
Okay, value lower than 1e-8 tends to cause non-convergence issue.

1e-7 works well.